src/main/java/net/runelite/launcher/Launcher.java

```diff
@@ -177,29 +189,54 @@ public static void main(String[] args)
 		catch (IOException | VerificationException | CertificateException | SignatureException | InvalidKeyException | NoSuchAlgorithmException ex)
 		{
 			log.error("error fetching bootstrap", ex);
-			frame.setVisible(false);
-			frame.dispose();
+			frame.errorMessage("Error fetching bootstrap");
+			frame.close();
 			System.exit(-1);
 			return;
 		}
 
+		frame.setMessage("Checking launcher version...", 10);
+		final String minLauncherVersion = bootstrap.getMinimumLauncherVersion();
+		final String launcherVersion = PROPERTIES.getVersion();
+		final Integer minJavaVersion = bootstrap.getMinimumJavaVersion();
+		final Integer javaVersion = getJavaFeatureVersion();
+		if ((minLauncherVersion != null && launcherVersion != null && minLauncherVersion.compareTo(launcherVersion) > 0)
+			|| (minJavaVersion != null && javaVersion != null && minJavaVersion < javaVersion))
+		{
+			try
+			{
+				SwingUtilities.invokeAndWait(frame::invalidVersion);
+				System.exit(0);

```

Comment:
This should be non-zero as well. Also, use positive exit codes everywhere, negative means system errors (like interruption of program or forced termination).